rmldatatfbbox/rmldatatfbbox/utils/helpers.py

```diff
@@ -0,0 +1,115 @@
+import os
+import json
+import cv2
+from pathlib import Path
+import tensorflow as tf
+from object_detection.utils import dataset_util
+
+### FUNCTIONS ###
+def construct_all(image_ids: list, temp_dir: Path):
+    labeled_images = {}
+    label_to_int_dict = {}
+
+    for image_id in image_ids:
+        labeled_images[image_id] = construct(image_id, temp_dir, label_to_int_dict)
+    
+    return labeled_images, label_to_int_dict
+
+def construct(image_id: str, temp_dir: Path, label_to_int_dict: dict):
+    if temp_dir is None:
+        cwd = Path.cwd()
+        data_dir = cwd / 'data'
+    else:
+        data_dir = temp_dir
+
+    image_filepath = None
+    image_type = None
+    file_extensions = [".png", ".jpg", ".jpeg"]
+    for extension in file_extensions:
+        if os.path.exists(data_dir / f'image_{image_id}{extension}'):
+            image_filepath = data_dir / f'image_{image_id}{extension}'
+            image_type = extension
+            ydim, xdim = tuple(cv2.imread(str(image_filepath.absolute())).shape[:2])
+            break
+
+    if image_filepath is None:
+        raise ValueError("Hmm, there doesn't seem to be a valid image filepath.")
+
+    with open(data_dir / f"meta_{image_id}.json", "r") as f:
+        meta = json.load(f)
+
+    label_boxes = []
+    for label, b in meta['bboxes'].items():
+        label_boxes.append({"label": label, "xmin": b['xmin'], "xmax": b['xmax'], "ymin": b['ymin'], "ymax": b['ymax']})
+        if label not in label_to_int_dict.keys():
+            label_to_int_dict[label] = len(label_to_int_dict)
+    
+    bboxes = {"image_id": image_id, "image_filepath": image_filepath, "image_type": image_type, "label_boxes": label_boxes, "xdim": xdim, "ydim": ydim}
```

Comment:
This is where the classes were used before, and the label_to_int_dict is now a parameter that is passed throughout the dataset creation process instead of being stored in a class